### ISSUE TITLE
Regression consistent-type-imports fixStyle not recognized on 5.33.1

### ISSUE THREAD

**Issue by zomars** [typescript-eslint/typescript-eslint#6481]
Saw this got solved on #5050 and released but it seems to be a regression here as I can replicate on latest version as of today.
```
Configuration for rule "@typescript-eslint/consistent-type-imports" is invalid:
Value {"fixStyle":"inline-type-imports"} should NOT have additional properties.
```

EDIT: 

Here's the relevant configuration:

```js
{
  rules: {
    "@typescript-eslint/consistent-type-imports": [
      "warn",
      { prefer: "type-imports", fixStyle: "inline-type-imports" },
    ],
  },
}
```

_Originally posted by @ysulyma in https://github.com/typescript-eslint/typescript-eslint/issues/5050#issuecomment-1312757001_

**Comment by JoshuaKGoldberg** [comment]
> 5.33.1
> latest version as of today

We're on [v5.52.0](https://www.npmjs.com/package/@typescript-eslint/eslint-plugin/v/5.52.0) now. I can't repro this issue on the playground: https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMAbXDYcAdwCG0JpAA0ipViXZIxaIgBmiaOijDEoiVJnbdByKcoAPAMpEkVyM0YmWxs7SWlUZwgAX2cAXUUYqKA

Please fill out the [appropriate issue template](https://github.com/typescript-eslint/typescript-eslint/issues/new/choose) when filing a new issue. They're there to help make sure all the needed information is filled out and to make sure you're on the latest version.
